types/src/error.rs

```diff
@@ -1,212 +1,279 @@
-use crate::consts::appconsts;
+use core::fmt::Display;
+
+use crate::types::String;
 
 /// Alias for a `Result` with the error type [`celestia_types::Error`].
 ///
 /// [`celestia_types::Error`]: crate::Error
-pub type Result<T, E = Error> = std::result::Result<T, E>;
+pub type Result<T, E = Error> = core::result::Result<T, E>;
 
 /// Representation of all the errors that can occur when interacting with [`celestia_types`].
 ///
 /// [`celestia_types`]: crate
-#[derive(Debug, thiserror::Error)]
+#[derive(Debug)]
```

Comment:
For all the error types that you removed `thiserror::Error` implement manually the `Error` trait if `std` feature is set. Otherwise our users will not be able to use our `Error` types with `anyhow` and `thiserror`.

```rust
#[cfg(feature = "std")]
impl std::error::Error for Error {}
```